federation/pkg/federation-controller/cluster/cluster_client.go

```diff
@@ -40,6 +40,32 @@ const (
 	KubeconfigSecretDataKey = "kubeconfig"
 )
 
+// This is to inject a different kubeconfigGetter in tests.
+// We dont use the standard one which calls NewInCluster in tests to avoid having to setup service accounts and mount files with secret tokens.
+var KubeconfigGetterForCluster = func(c *federation_v1alpha1.Cluster) clientcmd.KubeconfigGetter {
```

Comment:
It seems somewhat strange that this variable is manipulated/made test compatible by making it a global. It should probably be passed as an argument to `NewClusterClientSet()` or something like that. I understand that this isn't this PR/commit's problem but more about how this package is setup in general. It should probably be refactored in a future PR.